core/eolearn/core/utils/testing.py

```diff
@@ -79,4 +79,7 @@ def _get_feature_shape(
     rng: np.random.Generator, ftype: FeatureType, timestamps: List[dt.datetime], config: PatchGeneratorConfig
 ) -> Tuple[int, ...]:
     time, height, width, depth = len(timestamps), *config.raster_shape, rng.integers(*config.depth_range)
-    return (time, height, width, depth) if ftype.is_temporal() else (height, width, depth)
+    f_dim: int = ftype.ndim() or 0
+    if f_dim >= 3:
```

Comment:
so, this is the ugly part that we talked about and perhaps it would really make sense to change it as we discussed to:

```suggestion

    if ftype.is_spatial() and not ftype.is_vector():
```